test/distributed/test_c10d_common.py

```diff
@@ -1464,7 +1464,9 @@ def _call_collective_with_varying_tensors(self, backend, collective, *args):
         # ensure supported devices (cpu, cuda) succeeds during dispatch call
         tensor = torch.zeros(2, 2, device=torch.device(device))
         # multi tensor collectives
-        if collective == dist.all_gather:
+        if collective == dist.barrier:
+            collective()
+        elif collective == dist.all_gather:
             collective([tensor], tensor, *args)
```

Comment:
1. nit: this if-else block is getting bigger (as a result of wrapping/templating).
Maybe the test would be easier to read if we just write out each test call in _test_collectives, like:
```
dist.barrier()
dist.all_reduce(tensor)
...
```
2. Is `collective([tensor], tensor, *args)` a correct format for `all_gather`? i.e. it will have a list of only **one** tensor for the output. Or are we testing the dispatching functionality with WORLD_SIZE=1 here? (if so the code makes sense)

Reply:
1. Makes sense, the if statement is getting a bit messy, I will change it in future PRs.
2. We are just testing dispatching functionality and making sure the operation is callable. So handling the latter case, since collective correctness should be handled by other tests.